src/Promise.php

```diff
@@ -221,18 +221,36 @@ private function extract($promise)
 
     private function call(callable $callback)
     {
+        // Use reflection to inspect number of arguments expected by this callback.
+        // We did some careful benchmarking here: Using reflection to avoid unneeded
+        // function arguments is actually faster than blindly passing them.
```

Comment:
The reason for it being faster isn't avoiding the arguments being passed, but rather the three closure creations that are saved.

Reply:
@clue I think the motivation is completely covered by the circular references, the rest is just additional background on the performance of the patch. It's not super important to change, just a minor note I had while reading.